src/instructlab/model/list.py

```diff
@@ -0,0 +1,138 @@
+                model, age, size = _analyze_gguf(entry)
+                data.append([model, age, size])
+            elif entry.is_dir():
+                for m in _analyze_dir(entry, list_checkpoints, directory):
+                    data.append(m)
+    print_table(["Model Name", "Model Age", "Model Size"], data)
+
+
+# convert_bytes_to_proper_mag takes a dir/file size in Bytes and converts it to the proper Magnitude (MB, GB)
+def convert_bytes_to_proper_mag(f_size: float):
+    magnitudes = ["KB", "MB", "GB"]
+    magnitude = "B"
+    for mag in magnitudes:
+        if f_size >= 1024:
+            magnitude = mag
+            f_size /= 1024
+        else:
+            return f_size, magnitude
+    return f_size, magnitude
+
+
+# print_table displays the models on the system in a nicely formatted table
+def print_table(headers: list[str], data: list[list[str]]):
+    column_widths = [
+        max(len(str(row[i])) for row in data + [headers]) for i in range(len(headers))
+    ]
+    for i, header in enumerate(headers):
+        print(f"{header:{column_widths[i]}}", end=" | ")
+    print()
+
+    # Print separator line between headers and data
+    print("-" * (sum(column_widths) + 3 * len(headers) - 1))
+
+    for row in data:
+        for i, item in enumerate(row):
+            print(f"{item:{column_widths[i]}}", end=" | ")
+        print()
+
+
+AnalyzeResultGGUF = list[str]
+AnalyzeResultDir = list[list[str]]
+
+
+def _analyze_gguf(entry: Path) -> AnalyzeResultGGUF:
+    # stat the gguf, add it to the table
+    stat = Path(entry.absolute()).stat(follow_symlinks=False)
+    f_size = stat.st_size
+    f_size, magnitude = convert_bytes_to_proper_mag(f_size)
+    # add to table
+    age = pandas.to_timedelta(time.time_ns() - stat.st_mtime_ns)
+    return [entry.name, str(age), f"{f_size} {magnitude}"]
+
+
+def _analyze_dir(
+    entry: Path, list_checkpoints: bool, directory: str
+) -> AnalyzeResultDir:
+    actual_model_name = ""
+    all_files_sizes = 0
+    add_model = False
+    models = []
+    # walk entire dir.
+    for root, _, files in os.walk(entry.as_posix()):
+        normalized_path = os.path.normpath(root)
+        # Split the path into its components
+        parts = normalized_path.split(os.sep)
+        # Get the last two parts and join them back into a path
+        last_two_parts = os.path.join(parts[-2], parts[-1])
+        # if this is a dir it could be:
+        # top level repo dir `instructlab/`
+        # top level model dir `instructlab/granite-7b-lab`
+        # checkpoint top level dir `step-19`
+        # any lower level dir: `instructlab/granite-7b-lab/.huggingface/download.....`
+        # so, check if model is valid Safetensor, GGUF, or list it regardless w/ `--list-checkpoints`
+        # if --list-checkpoints is specified, we will list all checkpoints in the checkpoints dir regardless of the ModelCard
+        if is_model_safetensors(Path(normalized_path)) or is_model_gguf(
```

Comment:
please update comment to remove ModelCard reference on 114
Also wondering if we need the `is_model_gguf` check again here? seems redundant

Reply:
we need gguf check probably, there could be nested dirs.